### PR TITLE
patches: update patch release calendar for January 2022 cycle

### PR DESCRIPTION
**Pull request by xmudrii**
Update patch release calendar for January 2022 cycle.

/assign @saschagrunert @puerco @Verolop @cpanato

### CODE REVIEW

File: data/releases/schedule.yaml [hunk 1]
```diff
@@ -53,11 +65,14 @@ schedules:
       note: Regression https://groups.google.com/g/kubernetes-dev/c/KuF8s2zueFs
 - release: 1.20
   releaseDate: 2020-12-08
-  next: 1.20.14
-  cherryPickDeadline: 2021-12-10
-  targetDate: 2021-12-15
+  next: 1.20.15
```
Comment (xmudrii) [inline data/releases/schedule.yaml:68]: Do we want to add another 1.20 patch release to the calendar? If I'm not mistaken, 1.20 is now in the maintenance mode, so we should be doing patch releases for it only if it's needed (e.g. security fixes, critical bugs, etc.)?

cc @kubernetes/release-engineering

**Comment by xmudrii** [comment]
In addition to the question above, I'm also wondering how do we update https://kubernetes.io/releases/? It misses information about the latest releases.

File: data/releases/schedule.yaml (hunk 1 above)
Comment (cpanato) [reply data/releases/schedule.yaml:68]: we can schedule that and when is more closer to the day we decide

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**